github_app.go

```diff
@@ -149,6 +176,22 @@ func getInstallationID(ctx context.Context, authFields *GithubAppAuth, owner, re
 
 }
 
+func (g *GithubAppAuth) getInstallationToken(ctx context.Context, installationID int64, lifetime time.Duration, opts *github.InstallationTokenOptions) (string, error) {
```

Comment:
Is this helper function particularly helpful? Personally I think it reads easier to just have this logic in CreateCachedInstallationToken so it can be more easily compared with CreateInstallationToken. Or it seems like this and "createInstallationTokenForID" could just be combined into one helper function, and cache only if lifetime isn't 0 (or some cache boolean is true). (Having different public functions makes sense for ease of reading in base functions but I don't see the value for the private functions)